zenoh/src/net/runtime/orchestrator.rs

```diff
@@ -898,6 +898,26 @@ impl Runtime {
                                 tracing::trace!("Received {:?} from {}", msg.body, peer);
                                 if let ScoutingBody::Hello(hello) = &msg.body {
                                     if matcher.matches(hello.whatami) {
+
+                                        if let Ok(local_addr) = socket.local_addr() {
```

Comment:
I believe the right approach would be to return a list of `EndPoint` to connect to plus the additional metadata. This means the `HelloProto` data structure may not be the right data structure.

Reply:
It also means that https://github.com/eclipse-zenoh/zenoh/blob/e56a88b1dadbc33e515106c7f8f190746d071d85/zenoh/src/net/runtime/orchestrator.rs#L950 should accept `&[EndPoint]` instead of `&[Locator]`.

Reply:
Please have a look with the new code. Introducing the EndPoint aspect closer to the location, where Zenoh has the last time the chance to pick up the ethernet device information introduces quite a change.